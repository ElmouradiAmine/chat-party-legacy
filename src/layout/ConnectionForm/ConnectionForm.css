.connection-form {
  background-color: var(--color-primary);
  border: var(--border-primary);
  border-radius: 3px;

  display: flex;
  flex-direction: column;

  padding: 2.4rem;
}
.connection-form__username {
  margin-bottom: 3.2rem;
}
.connection-form__gender-selection {
  display: flex;
  justify-content: space-between;
  flex: 1;
  margin-bottom: 3.2rem;
}

.connection-form__gender-selection > *:not(:last-child) {
  margin-right: 3.2rem;
}
