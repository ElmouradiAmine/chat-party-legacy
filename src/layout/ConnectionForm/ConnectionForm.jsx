import React, { useState } from 'react';
import PropTypes from 'prop-types';

import './ConnectionForm.css';
import Input from '../../components/Input/Input';
import GenderMaleCard from '../../components/Card/GenderMaleCard/GenderMaleCard';
import GenderFemaleCard from '../../components/Card/GenderFemaleCard/GenderFemaleCard';

const ConnectionForm = ({ className }) => {
  const [username, setUsername] = useState('');

  const handleUsernameChange = (e) => {
    setUsername(e.target.value);
  };

  return (
    <form className={`connection-form ${className}`}>
      <Input
        type="text"
        className="connection-form__username"
        onChange={handleUsernameChange}
        value={username}
        placeholder="Username"
      />
      <div className="connection-form__gender-selection">
        <GenderMaleCard />
        <GenderFemaleCard />
      </div>
    </form>
  );
};

ConnectionForm.propTypes = {
  className: PropTypes.string,
};

ConnectionForm.defaultProps = {
  className: '',
};

export default ConnectionForm;
