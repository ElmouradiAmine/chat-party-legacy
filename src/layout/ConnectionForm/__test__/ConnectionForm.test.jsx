import React from 'react';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import ConnectionForm from '../ConnectionForm';

afterEach(cleanup);
describe('ConnectionForm component', () => {
  const className = 'className';
  const mockStore = configureStore();
  const initialState = {
    user: { user: null, error: '', status: 'idle' },
  };
  const store = mockStore(initialState);

  test('renders correctly', () => {
    render(
      <Provider store={store}>
        <ConnectionForm />
      </Provider>,
    );
  });

  test('renders correctly with props', () => {
    const { container } = render(
      <Provider store={store}>
        <ConnectionForm className={className} />
      </Provider>,
    );
    expect(container).not.toBeNull();
    expect(container.firstElementChild?.classList).toContain(className);
  });

  test('should control the input component value', () => {
    const { getByPlaceholderText } = render(
      <Provider store={store}>
        <ConnectionForm className={className} />
      </Provider>,
    );
    const input = getByPlaceholderText('Username');
    userEvent.type(input, 'test');
    expect(input).toHaveValue('test');
  });

  test('should control the gender cards component value', () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <ConnectionForm className={className} />
      </Provider>,
    );
    const genderCardMale = getByTestId('card-male');
    const genderCardFemale = getByTestId('card-female');

    expect(genderCardMale.classList).not.toContain('card--selected');
    expect(genderCardFemale.classList).not.toContain('card--selected');

    userEvent.click(genderCardMale);
    expect(genderCardMale.classList).toContain('card--selected');
    expect(genderCardFemale.classList).not.toContain('card--selected');

    userEvent.click(genderCardFemale);
    expect(genderCardMale.classList).not.toContain('card--selected');
    expect(genderCardFemale.classList).toContain('card--selected');
  });

  test('matches snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <ConnectionForm className={className} />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
