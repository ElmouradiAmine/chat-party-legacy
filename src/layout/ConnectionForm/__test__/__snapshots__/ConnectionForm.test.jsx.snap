// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectionForm component matches snapshot 1`] = `
<form
  className="connection-form className"
>
  <input
    className="input connection-form__username"
    onChange={[Function]}
    placeholder="Username"
    type="text"
    value=""
  />
  <div
    className="connection-form__gender-selection"
  >
    <div
      aria-checked={false}
      className="card  card--blue false "
      data-testid="card-male"
      id="card-male"
      role="radio"
      tabIndex={0}
    >
      <svg
        className="card__icon"
        data-testid="icon"
        fill={null}
      >
        gender-male-outline.svg
      </svg>
      <p
        className="card__label"
      >
        male
      </p>
    </div>
    <div
      aria-checked={false}
      className="card  card--pink false "
      data-testid="card-female"
      id="card-female"
      role="radio"
      tabIndex={0}
    >
      <svg
        className="card__icon"
        data-testid="icon"
        fill={null}
      >
        gender-female-outline.svg
      </svg>
      <p
        className="card__label"
      >
        female
      </p>
    </div>
  </div>
</form>
`;
